TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/WorkspacesController.php

```diff
@@ -395,7 +395,7 @@ public function publishWorkspaceAction(Workspace $workspace)
         if (($targetWorkspace = $workspace->getBaseWorkspace()) === null) {
             $targetWorkspace = $this->workspaceRepository->findOneByName('live');
         }
-        $workspace->publish($targetWorkspace);
+        $this->publishingService->publishNodes($this->publishingService->getUnpublishedNodes($workspace), $targetWorkspace);
```

Comment:
@aertmann just shortly discussed this with Karsten: the end result of both calls should be the same, but I really suggest to use `$workspace->publish()`.

Even though both variants currently load all unpublished nodes into memory and then publish them one-by-one, we have the option to optimize `$workspace->publish()` at some point and come up with a solution which is more memory efficient.

So, if there aren't any other reasons speaking against it, I'd rather use `$workspace->publish()` everywhere where we want to publish a whole workspace.